### PR TITLE
Avoid edge cases in activitylogs pipeline

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

Fixing pipeline breaking on some edge cases inside the azure activitylogs

## Why is it important?

>  There are some edge cases that result into failures for the ingest pipeline.
> 2.a) In my example I had \e characters in the log file which resulted into the error
> "Illegal unquoted character ((CTRL-CHAR, code 27)): has to be escaped using backslash to be included in string value\n at [Source: (org.elasticsearch.common.bytes.AbstractBytesReference$MarkSupportingStreamInputWrapper); line: 1, column: 2848]"
> 
> 2.b) Also the source.ip field is not always filled

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/beats/issues/20797

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -259,6 +259,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix long registry migration times. {pull}20717[20717] {issue}20705[20705]
 - Fix event types and categories in auditd module to comply with ECS {pull}20652[20652]
 - Update documentation in the azure module filebeat. {pull}20815[20815]
+- Add checks in the azure activitylogs pipeline for several edge cases. {issue}20797[20797]{pull}20865[20865]
```
Comment (jsoriano) [inline CHANGELOG.next.asciidoc:262]: Could you add some test files for these edge cases?

**Review by jsoriano — CHANGES_REQUESTED** [review]

**Comment by narph** [comment]
new PR coming soon, closing this since some of the issues have been fixed already
